R/item_deletion_h.R

```diff
@@ -0,0 +1,496 @@
+                                     paste0('\u0394', c("h(SP)")))
+   
+     rownames(h_R_Ef) <- rownames(h_str_vs_par_ref) <- rownames(h_str_vs_par_foc) <- 
+     c("h(TP)", "h(FP)", "h(TN)", "h(FN)", "h(PS)", "h(SR)", "h(SE)", "h(SP)")
+   
+  # Declare classes
+  store_par <- list(outputlist = store_par, condition = "partial",
+                    itemset = return_items)
+  store_str <- list(outputlist = store_str, condition = "strict", 
+                    itemset = return_items)
+  class(store_par) <- c("PartInvList", "PartInv")
+  class(store_str) <- c("PartInvList", "PartInv")
+  h_str_vs_par_list_ref <- list(outputlist = str_par_ref_list, condition="ref",
+                                itemset = return_items)
+  h_str_vs_par_list_foc <- list(outputlist = str_par_foc_list, condition="foc",
+                                itemset = return_items)
+  class(h_str_vs_par_list_ref) <-  c("PartInvList", "PartInv", "PartInv_groups" )
+  class(h_str_vs_par_list_foc) <-  c("PartInvList", "PartInv", "PartInv_groups")
+  
+  
+  # names(h_str_vs_par_ref) <- names(h_str_vs_par_foc) <- c("h(SFI, PFI)", 
+  #                                                         paste0("h(SFI, PFI|", c(1:N), c(")")))
+  names(h_str_vs_par_ref) <- names(h_str_vs_par_foc) <- c("SFI, PFI", 
+                                                          paste0("SFI, PFI|", c(1:N)))
+  
+  # names(delta_h_R_vs_Ef) <- paste0('\u0394', "h(r-Ef|", c(1:N), c(")"))
+  # names(delta_h_str_par_aggregate) <- paste0('\u0394', "h(SFI, PFI|", c(1:N), c(")"))
+  # names(h_aggregate_str_par) <- c("h(SFI, PFI)", paste0("h(SFI, PFI|", c(1:N), c(")")))
+  # 
+  names(delta_h_R_vs_Ef) <- paste0("r-Ef|", c(1:N))
+  names(delta_h_str_par_aggregate) <- paste0( "SFI, PFI|", c(1:N))
+  names(h_aggregate_str_par) <- c("SFI, PFI", paste0("SFI, PFI|", c(1:N)))
+  
+  
+  aggregate_par <- as.data.frame(cbind(aggregate_par))
+  h_aggregate_par <- as.data.frame(h_aggregate_par)
+  
+  h_str_vs_par <- list("ref"= t(h_str_vs_par_ref), "foc" = t(h_str_vs_par_foc))
+  
+  h_aggregate_str_par <- h_aggregate_str_par
+
+  returned <- list(
+    "ACAI" = t(aggregate_par),
+    "h ACAI (deletion)" = t(h_aggregate_par),
+    "h ACAI SFI-PFI" = t(h_aggregate_str_par),
+    "delta h ACAI SFI-PFI (deletion)" = t(delta_h_str_par_aggregate),
+    "AI Ratio" = t(AI_ratios),
+    "h CAI Ref-EF" = t(h_R_Ef),
+    "delta h CAI Ref-EF (deletion)" = t(delta_h_R_vs_Ef),
+    "h CAI SFI-PFI" = h_str_vs_par,
+    "delta h SFI-PFI (deletion)" = list("ref" = t(delta_h_str_vs_par_ref), 
+                                "foc" = t(delta_h_str_vs_par_foc)), 
+    "PartInv by groups" = list("reference" = h_str_vs_par_list_ref, 
+                     "focal" = h_str_vs_par_list_foc),
+    "PartInv" = list("strict" = store_str,
+                     "partial" = store_par),
+    "formatted" = print_formatted,
+    "return_items" = return_items)
+  class(returned) <- "itemdeletion"
+  
+  return(returned)
+}
```

Comment:
This doesn't need to be addressed in the pull request, but right now the function has 250 lines of code, and I feel like it may be helpful to move some of the components out as separate functions. For example, setting names and formatting.